Public/Src/Utilities/UnitTests/Executables/TestProcess/Operation.cs

```diff
@@ -288,6 +298,21 @@ private Operation(Type type, FileOrDirectoryArtifact? path = null, string conten
             AdditionalArgs = additionalArgs;
         }
 
+        private static Operation FromCommandLine(Type type, string path = null, string content = null, string linkPath = null, SymbolicLinkFlag? symLinkFlag = null, string additionalArgs = null)
+        {
+            Contract.Requires(content == null || !content.Contains(Environment.NewLine));
+
+            var result = new Operation(type: type,
+                content: content,
+                symLinkFlag: symLinkFlag,
+                doNotInfer: false,
+                additionalArgs: additionalArgs);
+            result.PathAsString = path;
+            result.LinkPathAsString = linkPath;

```

Comment:
nit: should we create another ctor that takes strings?

Reply:
I preferred a static helper method over a constructor here in order to give it a name to make it clear in what context it should be used. A constructor would have made it appear more general purpose.